### DISCUSSION TITLE
Combining/deciding labels in grouped updates

### DISCUSSION THREAD

**Discussion by narwold** [renovatebot/renovate#8887] [Request Help]
In a monorepo where I have mixed update types that are needed:
- webpack 4.x.y in one package
- webpack 5.x.y in another package

I've tried setting `separateMajorMinor` to both true and false, and in both cases a PR is opened that updates both to the latest version of 5.x.y. This part is what I want. What I don't understand is that with this config...

```
  "packageRules": [
    {
      "matchUpdateTypes": ["major"],
      "labels": ["dependencies", "UPDATE-MAJOR"]
    }
  ],
```

...Renovate seems to NOT treat this PR as a major bump, which it should be. If I add a minor bump rule, it will label it that, but nothing I do seems to get Renovate to recognize this mixed PR as a major bump.

Presumably this means if I turn on auto-merge it could also potentially auto-merge something with a major bump in it just because it ALSO happens to have a minor bump, even if I tell it not to automerge majors? This is concerning to me.

**Comment by rarkins** [comment]
There is most likely an error in the configuration or some other misunderstanding. 

Eg labels get merged in a multi-update PR but automerge will only happen if all updates are true. 

If you can demonstrate each problem in a public repository then we can help you find an answer.

**Reply by narwold** [reply]
PR: https://github.com/beamtech/renovate-test/pull/22
config: https://github.com/beamtech/renovate-test/blob/master/renovate.json

**Reply by narwold** [reply]
If it can't take the most significant bump and label based off of that, then labels are useless to us, because the `UPDATE-MAJOR` label won't show us all our major bump PRs - only ones that don't also have minors in them.

**Comment by rarkins** [comment]
I found something really old in our code which prevents merging of labels, hoping to fix it here: #8892

**Comment by narwold** [comment]
That appears to have fixed it. Haven't tried auto-merge yet, but I'll assume good things until I do. Thanks!
